operator/validator/controller.go

```diff
@@ -945,36 +935,39 @@ func (c *controller) onShareInit(share *ssvtypes.SSVShare) (*validators.Validato
 		c.printShare(v.Share(), "get validator")
 	}
 
+	var foundVC *validator.Committee
+
 	// Start a committee validator.
-	vc, found := c.validatorsMap.GetCommittee(operator.CommitteeID)
-	if !found {
-		// Share context with both the validator and the runners,
-		// so that when the validator is stopped, the runners are stopped as well.
-		ctx, cancel := context.WithCancel(c.ctx)
+	if c.validatorsMap.UpdateCommitteeAtomic(operator.CommitteeID, func(vc *validator.Committee) {
+		vc.AddShare(&share.Share)
+		c.printShare(share, "added share to committee")
+		foundVC = vc
+	}) {
+		return v, foundVC, nil
+	}
 
-		opts := c.validatorOptions
-		opts.SSVShare = share
-		opts.Operator = operator
+	// Share context with both the validator and the runners,
+	// so that when the validator is stopped, the runners are stopped as well.
+	ctx, cancel := context.WithCancel(c.ctx)
 
-		committeeOpIDs := types.OperatorIDsFromOperators(operator.Committee)
+	opts := c.validatorOptions
+	opts.SSVShare = share
+	opts.Operator = operator
 
-		logger := c.logger.With([]zap.Field{
-			zap.String("committee", fields.FormatCommittee(committeeOpIDs)),
-			zap.String("committee_id", hex.EncodeToString(operator.CommitteeID[:])),
-		}...)
+	committeeOpIDs := types.OperatorIDsFromOperators(operator.Committee)
 
-		committeeRunnerFunc := SetupCommitteeRunners(ctx, opts)
+	logger := c.logger.With([]zap.Field{
+		zap.String("committee", fields.FormatCommittee(committeeOpIDs)),
+		zap.String("committee_id", hex.EncodeToString(operator.CommitteeID[:])),
+	}...)
```

Comment:
it works without explicitly using `zap.Field` 
```suggestion
	logger := c.logger.With(
		zap.String("committee", fields.FormatCommittee(committeeOpIDs)),
		zap.String("committee_id", hex.EncodeToString(operator.CommitteeID[:])))
```